### PR TITLE
Fixed common code to use inbuilt date parser

### PR DESCRIPTION
**Pull request by abeshi-softwire**

### CODE REVIEW

File: SharedCode/src/commonMain/kotlin/common.kt [hunk 1]
```diff
@@ -41,13 +43,9 @@ fun deserialiseJson(jsonString: String): ApiResult {
...
-    // NOTE: Not completely accurate
-    // TODO: make better?
-    val year: Long = s.substring(0,4).toLong()
-    val month: Long = s.substring(5,7).toLong()
-    val day: Long = s.substring(8,10).toLong()
-    val hour: Long = s.substring(11,13).toLong()
-    return 365*24*60*60*(year-1970) + 31*24*60*60*(month-1) + 24*60*60*day + 60*60*hour
+    val formatter = DateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSz")
+    val date = formatter.parse(s)
+    return date.utc.unixMillisLong
```
Comment (thevirtuoso1973) [inline SharedCode/src/commonMain/kotlin/common.kt:48]: Should we return the time in seconds? That's usually what I think of when working in Epoch time. Or we could change the function name to getEpochMillisecondsFromUTC.

**Review by thevirtuoso1973 — CHANGES_REQUESTED** [review]
See common.kt file for comment.

File: SharedCode/src/commonMain/kotlin/common.kt (hunk 1 above)
Comment (abeshi-softwire) [reply SharedCode/src/commonMain/kotlin/common.kt:48]: I don't know. We should certainly stick to one in the common code, I guess now is the time to decide.

Comment (abeshi-softwire) [reply SharedCode/src/commonMain/kotlin/common.kt:48]: I believe so far we're using milliseconds

Comment (thevirtuoso1973) [reply SharedCode/src/commonMain/kotlin/common.kt:48]: The function returned it in seconds before your change, unless my arithmetic is off.

Comment (abeshi-softwire) [reply SharedCode/src/commonMain/kotlin/common.kt:48]: What do you return in your android getCurrentUnixTime?

Comment (abeshi-softwire) [reply SharedCode/src/commonMain/kotlin/common.kt:48]: I return milliseconds from iOS

Comment (abeshi-softwire) [reply SharedCode/src/commonMain/kotlin/common.kt:48]: If you need to make some changes to the Android code feel free to put them on this branch first, we can look them over and then merge it all at once

Comment (thevirtuoso1973) [reply SharedCode/src/commonMain/kotlin/common.kt:48]: Okay looks like I also return milliseconds.

**Pull request merged**